sdk/eventhub/Azure.Messaging.EventHubs/CONTRIBUTING.md

```diff
@@ -1,19 +1,70 @@
+### Running tests
+
+The Event Hubs client library tests may be executed using the `dotnet` CLI, or the test runner of your choice - such as Visual Studio or Visual Studio Code.  For those developers using Visual Studio, it is safe to use the Live Unit Testing feature, as any tests with external dependencies have been marked to be excluded.
+
+Tests in the Event Hubs client library are split into two categories - unit tests and integration tests.  For unit tests, there are no special considerations; these are self-contained and execute locally without any reliance on external resources.  Unit tests are considered the default test type in the Event Hubs client library and, thus, have no explicit category trait attached to them.  
+
+Integration tests, have dependencies on live Azure resources and require setting up your development environment prior to running.  Known in the Azure SDK project commonly as "Live" tests, these tests are decorated with a category trait of "Live".  Specifically, an Azure resource group, Event Hubs namespace, and Azure Service Principal with "contributor" rights to the Event Hub namespace is required.   The Live tests read information from the following environment variables:
+
+ `EVENT_HUBS_CONNECTION_STRING`  
+  The full connection string to the Event Hubs namespace, using the default shared access policy.
+    
+`EVENT_HUBS_NAMESPACE`  
+ The host name of the Event Hubs namespace, without the protocol or domain found in the connection string.
+    
+`EVENT_HUBS_RESOURCEGROUP`  
+ The name of the Azure resource group that contains the Event Hubs namespace
+   
+`EVENT_HUBS_SUBSCRIPTION`  
+ The identifier (GUID) of the Azure subscription to which the service principal belongs
+    
+`EVENT_HUBS_TENANT`  
+ The identifier (GUID) of the Azure Active Directory tenant that contains the service principal
+
+`EVENT_HUBS_CLIENT`  
+ The identifier (GUID) of the Azure Active Directory application that is associated with the service principal
+   
+`EVENT_HUBS_SECRET`  
+ The client secret (password) of the Azure Active Directory application that is associated with the service principal
+   
+To make setting up your environment easier, a [PowerShell script](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1) is included in the repository and will create and/or configure the needed Azure resources.  To use this script, open a PowerShell instance and login to your Azure account using `Login-AzAccount`, then execute the script.   You will need to provide some information, after which the script will configure the Azure resources and then output the set of environment variables with the correct values for running tests.
+
```

Comment:
3 spaces before 'You will need'

Reply:
Why the heck do I keep doing that?   This is like the 5th one...

Reply:
I think your space bar got excited